
1:16 PM 12/14/2017


cody@LAPTOP-TO4068AG MINGW64 /
$ cd gitcode/1USEthis2017Oct25

cody@LAPTOP-TO4068AG MINGW64 /gitcode/1USEthis2017Oct25
$ ls
1aHowTO/  2HOWto/                   balletclassstu/  gymtime/    spellingbee/
1HowTO/   2myExptpythonJSgmailAPI/  goodmed/         merrymail/  sqlmail/

cody@LAPTOP-TO4068AG MINGW64 /gitcode/1USEthis2017Oct25
$ cd sqlmail

cody@LAPTOP-TO4068AG MINGW64 /gitcode/1USEthis2017Oct25/sqlmail (master)
$ ls
__pycache__/        main.py        readme.txt          sqlmailhome.html
client_secret.json  maingym.py     sqlmailbye.html     templates/
emailer.py          quickstart.py  sqlmailemail2.html

cody@LAPTOP-TO4068AG MINGW64 /gitcode/1USEthis2017Oct25/sqlmail (master)
$ conda env list
# conda environments:
#
env_flask                C:\Miniconda3\envs\env_flask
envindia_flask           C:\Miniconda3\envs\envindia_flask
hello-flask              C:\Miniconda3\envs\hello-flask
helloflask               C:\Miniconda3\envs\helloflask
root                  *  C:\Miniconda3


cody@LAPTOP-TO4068AG MINGW64 /gitcode/1USEthis2017Oct25/sqlmail (master)
$ source activate env_flask
(env_flask)
cody@LAPTOP-TO4068AG MINGW64 /gitcode/1USEthis2017Oct25/sqlmail (master)
$ python maingym.py
 * Restarting with stat
 * Debugger is active!
 * Debugger PIN: 104-362-096
 * Running on http://127.0.0.1:5000/ (Press CTRL+C to quit)
127.0.0.1 - - [14/Dec/2017 13:19:19] "GET / HTTP/1.1" 200 -
127.0.0.1 - - [14/Dec/2017 13:19:31] "GET /email HTTP/1.1" 200 -
forProj
gapp2017falloct@gmail.com
13
127.0.0.1 - - [14/Dec/2017 13:19:47] "POST /email HTTP/1.1" 500 -
Traceback (most recent call last):
  File "C:\Miniconda3\envs\env_flask\lib\site-packages\flask\app.py", line 1997,
 in __call__
    return self.wsgi_app(environ, start_response)
  File "C:\Miniconda3\envs\env_flask\lib\site-packages\flask\app.py", line 1985,
 in wsgi_app
    response = self.handle_exception(e)
  File "C:\Miniconda3\envs\env_flask\lib\site-packages\flask\app.py", line 1540,
 in handle_exception
    reraise(exc_type, exc_value, tb)
  File "C:\Miniconda3\envs\env_flask\lib\site-packages\flask\_compat.py", line 3
3, in reraise
    raise value
  File "C:\Miniconda3\envs\env_flask\lib\site-packages\flask\app.py", line 1982,
 in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Miniconda3\envs\env_flask\lib\site-packages\flask\app.py", line 1614,
 in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Miniconda3\envs\env_flask\lib\site-packages\flask\app.py", line 1517,
 in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "C:\Miniconda3\envs\env_flask\lib\site-packages\flask\_compat.py", line 3
3, in reraise
    raise value
  File "C:\Miniconda3\envs\env_flask\lib\site-packages\flask\app.py", line 1612,
 in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Miniconda3\envs\env_flask\lib\site-packages\flask\app.py", line 1598,
 in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "C:\Users\cody\Downloads\Git\gitcode\1USEthis2017Oct25\sqlmail\maingym.py
", line 53, in email_post
    send_email(email,firstname,exerhours)
  File "C:\Users\cody\Downloads\Git\gitcode\1USEthis2017Oct25\sqlmail\emailer.py
", line 42, in send_email
    message = messages.send(userId='me', body=body).execute()
  File "C:\Miniconda3\envs\env_flask\lib\site-packages\oauth2client\_helpers.py"
, line 133, in positional_wrapper
    return wrapped(*args, **kwargs)
  File "C:\Miniconda3\envs\env_flask\lib\site-packages\googleapiclient\http.py",
 line 842, in execute
    raise HttpError(resp, content, uri=self.uri)
googleapiclient.errors.HttpError: <HttpError 403 when requesting https://www.goo
gleapis.com/gmail/v1/users/me/messages/send?alt=json returned "Insufficient Perm
ission">
127.0.0.1 - - [14/Dec/2017 13:19:47] "GET /email?__debugger__=yes&cmd=resource&f
=style.css HTTP/1.1" 200 -
127.0.0.1 - - [14/Dec/2017 13:19:47] "GET /email?__debugger__=yes&cmd=resource&f
=jquery.js HTTP/1.1" 200 -
127.0.0.1 - - [14/Dec/2017 13:19:47] "GET /email?__debugger__=yes&cmd=resource&f
=debugger.js HTTP/1.1" 200 -
127.0.0.1 - - [14/Dec/2017 13:19:47] "GET /email?__debugger__=yes&cmd=resource&f
=ubuntu.ttf HTTP/1.1" 200 -
127.0.0.1 - - [14/Dec/2017 13:19:47] "GET /email?__debugger__=yes&cmd=resource&f
=console.png HTTP/1.1" 200 -
(env_flask)
cody@LAPTOP-TO4068AG MINGW64 /gitcode/1USEthis2017Oct25/sqlmail (master)
$ cd
(env_flask)
cody@LAPTOP-TO4068AG MINGW64 ~
$ ls -a
 ./
 ../
 .android/
 .atom/
 .bash_history
 .bash_profile
 .bashrc
 .conda/
 .credentials/
 .eclipse/
 .gitconfig
 .idlerc/
 .minttyrc
 .p2/
 .ssh/
 .tooling/
 .viminfo
'3D Objects'/
 AppData/
'Application Data'@
 code/
 Contacts/
 Cookies@
 Desktop/
 Documents/
 Downloads/
 Favorites/
 git/
 Google/
 IntelGraphicsProfiles/
 Links/
'Local Settings'@
 mercurial.ini
 MicrosoftEdgeBackups/
 Music/
'My Documents'@
 NetHood@
 NTUSER.DAT
 ntuser.dat.LOG1
 ntuser.dat.LOG2
 NTUSER.DAT{e259a95a-cbbd-11e7-bb13-a81e841a73e0}.TM.blf
 NTUSER.DAT{e259a95a-cbbd-11e7-bb13-a81e841a73e0}.TMContainer0000000000000000000                         1.regtrans-ms
 NTUSER.DAT{e259a95a-cbbd-11e7-bb13-a81e841a73e0}.TMContainer0000000000000000000                         2.regtrans-ms
 ntuser.ini
 OneDrive/
 PicStream/
 Pictures/
 PrintHood@
 Recent@
'Saved Games'/
 Searches/
 SendTo@
'Start Menu'@
 Templates@
 Videos/
 workspace/
(env_flask)
cody@LAPTOP-TO4068AG MINGW64 ~
$ cd .credentials
(env_flask)
cody@LAPTOP-TO4068AG MINGW64 ~/.credentials
$ ls -a
./  ../  gmail-python-quickstart.json
(env_flask)
cody@LAPTOP-TO4068AG MINGW64 ~/.credentials
$ rm gmail*
(env_flask)
cody@LAPTOP-TO4068AG MINGW64 ~/.credentials
$ cd
(env_flask)
cody@LAPTOP-TO4068AG MINGW64 ~
$ cd gitcode/1USEthis2017Oct25
bash: cd: gitcode/1USEthis2017Oct25: No such file or directory
(env_flask)
cody@LAPTOP-TO4068AG MINGW64 ~
$ cd ..
(env_flask)
cody@LAPTOP-TO4068AG MINGW64 /c/Users
$ ls
'All Users'@   Default/         Default.migrated/   hannah/
 cody/        'Default User'@   desktop.ini         Public/
(env_flask)
cody@LAPTOP-TO4068AG MINGW64 /c/Users
$ cd gitcode
bash: cd: gitcode: No such file or directory
(env_flask)
cody@LAPTOP-TO4068AG MINGW64 /c/Users
$ ls
'All Users'@   Default/         Default.migrated/   hannah/
 cody/        'Default User'@   desktop.ini         Public/
(env_flask)
cody@LAPTOP-TO4068AG MINGW64 /c/Users
$ cd cody
(env_flask)
cody@LAPTOP-TO4068AG MINGW64 ~
$ ls
'3D Objects'/
 AppData/
'Application Data'@
 code/
 Contacts/
 Cookies@
 Desktop/
 Documents/
 Downloads/
 Favorites/
 git/
 Google/
 IntelGraphicsProfiles/
 Links/
'Local Settings'@
 mercurial.ini
 MicrosoftEdgeBackups/
 Music/
'My Documents'@
 NetHood@
 NTUSER.DAT
 ntuser.dat.LOG1
 ntuser.dat.LOG2
 NTUSER.DAT{e259a95a-cbbd-11e7-bb13-a81e841a73e0}.TM.blf
 NTUSER.DAT{e259a95a-cbbd-11e7-bb13-a81e841a73e0}.TMContainer0000000000000000000                         1.regtrans-ms
 NTUSER.DAT{e259a95a-cbbd-11e7-bb13-a81e841a73e0}.TMContainer0000000000000000000                         2.regtrans-ms
 ntuser.ini
 OneDrive/
 PicStream/
 Pictures/
 PrintHood@
 Recent@
'Saved Games'/
 Searches/
 SendTo@
'Start Menu'@
 Templates@
 Videos/
 workspace/
(env_flask)
cody@LAPTOP-TO4068AG MINGW64 ~
$ cd Downloads/Git/gitcode/1USEthis2017Oct25/sqlmail
(env_flask)
cody@LAPTOP-TO4068AG MINGW64 ~/Downloads/Git/gitcode/1USEthis2017Oct25/sqlmail (                         master)
$ ls
__pycache__/        main.py        readme.txt          sqlmailhome.html
client_secret.json  maingym.py     sqlmailbye.html     templates/
emailer.py          quickstart.py  sqlmailemail2.html
(env_flask)
cody@LAPTOP-TO4068AG MINGW64 ~/Downloads/Git/gitcode/1USEthis2017Oct25/sqlmail (                         master)
$ python maingym.py
 * Restarting with stat
 * Debugger is active!
 * Debugger PIN: 104-362-096
 * Running on http://127.0.0.1:5000/ (Press CTRL+C to quit)
127.0.0.1 - - [14/Dec/2017 13:24:24] "GET / HTTP/1.1" 200 -
127.0.0.1 - - [14/Dec/2017 13:24:31] "GET /email HTTP/1.1" 200 -
forProj
gapp2017falloct@gmail.com
11
C:\Miniconda3\envs\env_flask\lib\site-packages\oauth2client\_helpers.py:255: Use
rWarning: Cannot access C:\Users\cody\.credentials\gmail-python-quickstart.json:
 No such file or directory
  warnings.warn(_MISSING_FILE_MESSAGE.format(filename))

Your browser has been opened to visit:

    https://accounts.google.com/o/oauth2/auth?client_id=1021640295565-0rhctbbeb2
668laqnoseie4dcupd2gur.apps.googleusercontent.com&redirect_uri=http%3A%2F%2Floca
lhost%3A8080%2F&scope=https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fgmail.send&acces
s_type=offline&response_type=code

If your browser is on a different machine then exit and re-run this
application with the command-line parameter

  --noauth_local_webserver

Authentication successful.
Storing credentials to C:\Users\cody\.credentials\gmail-python-quickstart.json
127.0.0.1 - - [14/Dec/2017 13:25:06] "POST /email HTTP/1.1" 500 -
Traceback (most recent call last):
  File "C:\Miniconda3\envs\env_flask\lib\site-packages\flask\app.py", line 1997,
 in __call__
    return self.wsgi_app(environ, start_response)
  File "C:\Miniconda3\envs\env_flask\lib\site-packages\flask\app.py", line 1985,
 in wsgi_app
    response = self.handle_exception(e)
  File "C:\Miniconda3\envs\env_flask\lib\site-packages\flask\app.py", line 1540,
 in handle_exception
    reraise(exc_type, exc_value, tb)
  File "C:\Miniconda3\envs\env_flask\lib\site-packages\flask\_compat.py", line 3
3, in reraise
    raise value
  File "C:\Miniconda3\envs\env_flask\lib\site-packages\flask\app.py", line 1982,
 in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Miniconda3\envs\env_flask\lib\site-packages\flask\app.py", line 1614,
 in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Miniconda3\envs\env_flask\lib\site-packages\flask\app.py", line 1517,
 in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "C:\Miniconda3\envs\env_flask\lib\site-packages\flask\_compat.py", line 3
3, in reraise
    raise value
  File "C:\Miniconda3\envs\env_flask\lib\site-packages\flask\app.py", line 1612,
 in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Miniconda3\envs\env_flask\lib\site-packages\flask\app.py", line 1598,
 in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "C:\Users\cody\Downloads\Git\gitcode\1USEthis2017Oct25\sqlmail\maingym.py
", line 53, in email_post
    send_email(email,firstname,exerhours)
  File "C:\Users\cody\Downloads\Git\gitcode\1USEthis2017Oct25\sqlmail\emailer.py
", line 42, in send_email
    message = messages.send(userId='me', body=body).execute()
  File "C:\Miniconda3\envs\env_flask\lib\site-packages\oauth2client\_helpers.py"
, line 133, in positional_wrapper
    return wrapped(*args, **kwargs)
  File "C:\Miniconda3\envs\env_flask\lib\site-packages\googleapiclient\http.py",
 line 842, in execute
    raise HttpError(resp, content, uri=self.uri)
googleapiclient.errors.HttpError: <HttpError 403 when requesting https://www.goo
gleapis.com/gmail/v1/users/me/messages/send?alt=json returned "Access Not Config
ured. Gmail API has not been used in project 1021640295565 before or it is disab
led. Enable it by visiting https://console.developers.google.com/apis/api/gmail.
googleapis.com/overview?project=1021640295565 then retry. If you enabled this AP
I recently, wait a few minutes for the action to propagate to our systems and re
try.">
127.0.0.1 - - [14/Dec/2017 13:25:06] "GET /email?__debugger__=yes&cmd=resource&f
=jquery.js HTTP/1.1" 200 -
127.0.0.1 - - [14/Dec/2017 13:25:06] "GET /email?__debugger__=yes&cmd=resource&f
=style.css HTTP/1.1" 200 -
127.0.0.1 - - [14/Dec/2017 13:25:06] "GET /email?__debugger__=yes&cmd=resource&f
=debugger.js HTTP/1.1" 200 -
127.0.0.1 - - [14/Dec/2017 13:25:06] "GET /email?__debugger__=yes&cmd=resource&f
=ubuntu.ttf HTTP/1.1" 200 -
127.0.0.1 - - [14/Dec/2017 13:25:06] "GET /email?__debugger__=yes&cmd=resource&f
=console.png HTTP/1.1" 200 -
127.0.0.1 - - [14/Dec/2017 13:26:50] "GET / HTTP/1.1" 200 -
127.0.0.1 - - [14/Dec/2017 13:26:58] "GET /email HTTP/1.1" 200 -
forProj
gapp2017falloct@gmail.com
65
127.0.0.1 - - [14/Dec/2017 13:27:10] "POST /email HTTP/1.1" 200 -
 * Detected change in 'C:\\Users\\cody\\Downloads\\Git\\gitcode\\1USEthis2017Oct
25\\sqlmail\\maingym.py', reloading
 * Restarting with stat
 * Debugger is active!
 * Debugger PIN: 104-362-096
 * Running on http://127.0.0.1:5000/ (Press CTRL+C to quit)
 * Detected change in 'C:\\Users\\cody\\Downloads\\Git\\gitcode\\1USEthis2017Oct
25\\sqlmail\\maingym.py', reloading
 * Restarting with stat
 * Debugger is active!
 * Debugger PIN: 104-362-096
 * Running on http://127.0.0.1:5000/ (Press CTRL+C to quit)
 * Detected change in 'C:\\Users\\cody\\Downloads\\Git\\gitcode\\1USEthis2017Oct
25\\sqlmail\\maingym.py', reloading
 * Restarting with stat
 * Debugger is active!
 * Debugger PIN: 104-362-096
 * Running on http://127.0.0.1:5000/ (Press CTRL+C to quit)
127.0.0.1 - - [14/Dec/2017 13:34:32] "GET /exerhours HTTP/1.1" 200 -
127.0.0.1 - - [14/Dec/2017 13:34:33] "GET /sqlmailstyling/sqlmailstyle2.css HTTP
/1.1" 404 -
 * Detected change in 'C:\\Users\\cody\\Downloads\\Git\\gitcode\\1USEthis2017Oct
25\\sqlmail\\maingym.py', reloading
 * Restarting with stat
 * Debugger is active!
 * Debugger PIN: 104-362-096
 * Running on http://127.0.0.1:5000/ (Press CTRL+C to quit)
127.0.0.1 - - [14/Dec/2017 13:41:26] "GET /exerhours HTTP/1.1" 200 -
127.0.0.1 - - [14/Dec/2017 13:41:26] "GET /static/sqlmailstyle1.css HTTP/1.1" 20
0 -
127.0.0.1 - - [14/Dec/2017 13:47:40] "GET /exerhours HTTP/1.1" 200 -
127.0.0.1 - - [14/Dec/2017 13:47:40] "GET /static/sqlmailstyle1.css HTTP/1.1" 30
4 -
 * Detected change in 'C:\\Users\\cody\\Downloads\\Git\\gitcode\\1USEthis2017Oct
25\\sqlmail\\maingym.py', reloading
 * Restarting with stat
 * Debugger is active!
 * Debugger PIN: 104-362-096
 * Running on http://127.0.0.1:5000/ (Press CTRL+C to quit)
127.0.0.1 - - [14/Dec/2017 13:50:29] "POST /exerhours HTTP/1.1" 405 -
 * Detected change in 'C:\\Users\\cody\\Downloads\\Git\\gitcode\\1USEthis2017Oct
25\\sqlmail\\maingym.py', reloading
 * Restarting with stat
Traceback (most recent call last):
  File "maingym.py", line 13, in <module>
    @app.route("/exerhours", method="GET")
  File "C:\Miniconda3\envs\env_flask\lib\site-packages\flask\app.py", line 1080,
 in decorator
    self.add_url_rule(rule, endpoint, f, **options)
  File "C:\Miniconda3\envs\env_flask\lib\site-packages\flask\app.py", line 64, i
n wrapper_func
    return f(self, *args, **kwargs)
  File "C:\Miniconda3\envs\env_flask\lib\site-packages\flask\app.py", line 1043,
 in add_url_rule
    rule = self.url_rule_class(rule, methods=methods, **options)
TypeError: __init__() got an unexpected keyword argument 'method'
(env_flask)
cody@LAPTOP-TO4068AG MINGW64 ~/Downloads/Git/gitcode/1USEthis2017Oct25/sqlmail (                         master)
$ python maingym.py
  File "maingym.py", line 12
    @app route("/exerhours")
             ^
SyntaxError: invalid syntax
(env_flask)
cody@LAPTOP-TO4068AG MINGW64 ~/Downloads/Git/gitcode/1USEthis2017Oct25/sqlmail (                         master)
$ python maingym.py
  File "maingym.py", line 12
    @app route("/exerhours",methods=["GET"])
             ^
SyntaxError: invalid syntax
(env_flask)
cody@LAPTOP-TO4068AG MINGW64 ~/Downloads/Git/gitcode/1USEthis2017Oct25/sqlmail (                         master)
$ python maingym.py
  File "maingym.py", line 12
    @app route("/exerhours")
             ^
SyntaxError: invalid syntax
(env_flask)
cody@LAPTOP-TO4068AG MINGW64 ~/Downloads/Git/gitcode/1USEthis2017Oct25/sqlmail (                         master)
$ python maingym.py
Traceback (most recent call last):
  File "maingym.py", line 15, in <module>
    @app.route("/exerhours",methods="POST")
  File "C:\Miniconda3\envs\env_flask\lib\site-packages\flask\app.py", line 1080,
 in decorator
    self.add_url_rule(rule, endpoint, f, **options)
  File "C:\Miniconda3\envs\env_flask\lib\site-packages\flask\app.py", line 64, i
n wrapper_func
    return f(self, *args, **kwargs)
  File "C:\Miniconda3\envs\env_flask\lib\site-packages\flask\app.py", line 1021,
 in add_url_rule
    raise TypeError('Allowed methods have to be iterables of strings, '
TypeError: Allowed methods have to be iterables of strings, for example: @app.ro
ute(..., methods=["POST"])
(env_flask)
cody@LAPTOP-TO4068AG MINGW64 ~/Downloads/Git/gitcode/1USEthis2017Oct25/sqlmail (                         master)
$ python maingym.py
Traceback (most recent call last):
  File "maingym.py", line 15, in <module>
    @app.route("/exerhours",methods=["POST"])
  File "C:\Miniconda3\envs\env_flask\lib\site-packages\flask\app.py", line 1080, in decorator
    self.add_url_rule(rule, endpoint, f, **options)
  File "C:\Miniconda3\envs\env_flask\lib\site-packages\flask\app.py", line 64, in wrapper_func
    return f(self, *args, **kwargs)
  File "C:\Miniconda3\envs\env_flask\lib\site-packages\flask\app.py", line 1051, in add_url_rule
    'existing endpoint function: %s' % endpoint)
AssertionError: View function mapping is overwriting an existing endpoint function: exerhours
(env_flask)
cody@LAPTOP-TO4068AG MINGW64 ~/Downloads/Git/gitcode/1USEthis2017Oct25/sqlmail (master)
$
